packages/extensions/modelerfour/src/modeler/modelerfour.ts

```diff
@@ -2125,8 +2125,27 @@ export class ModelerFour {
   processResponses(httpOperation: OpenAPI.HttpOperation, operation: Operation) {
     const acceptTypes = new Set<string>();
 
+    const modelerResponses = httpOperation.responses;
+
+    // If the operation has final-state-schema lro option, add a "200" response if there isn't one already
+    if (httpOperation["x-ms-long-running-operation-options"]?.["final-state-schema"]) {
+      const finalStateSchema = httpOperation["x-ms-long-running-operation-options"]?.["final-state-schema"] as string;
+      if (!Object.keys(modelerResponses).some((k) => k === "200")) {
+        modelerResponses["200"] = {
+          description: "Success",
+          content: {
+            "application/json": {
+              schema: {
+                $ref: finalStateSchema,
```

Comment:
Have you tested that from e2e including in the swagger -> openapi3 conversion?

I think for that change we need to make sure the $ref gets converted in every plugin(it might just work, because a lot will just recursively look for $ref)

Reply:
For swagger -> openapi3 I think all we want is for the extension to be carried forward faithfully ... and it appears that is the case.  It does not -- and IMHO should not -- manufacture a "fake" 200 response in the openapi3.  I'm hoping that we do not need to do that in order for the SDK generators to act as if a 200 response was present.  How can we verify that?

Reply:
Ah! I understand better now. Yes ... we need to fix the ref. I will try your suggestion to see how that works.